src/state/actions/auth.ts

```diff
@@ -182,37 +190,90 @@ const syncItemsWithLocalStorage = async (userInfo: any) => {
 
 export const loginUser = async (credentials: Credentials) => {
   try {
-    const userInfo = await login(credentials.email, credentials.password)
     const globalState = getGlobal()
+    const localUserInfo = globalState.session.userInfo
+    const serverUserInfo = await login(credentials.email, credentials.password)
     const { valueTagSettings } = globalState.session
 
     const localFCMSaved = await fcmTokenGetLocally()
-    userInfo.notificationsEnabled = !!localFCMSaved
+    serverUserInfo.notificationsEnabled = !!localFCMSaved
 
-    setGlobal({
-      session: {
-        userInfo,
-        isLoggedIn: true,
-        valueTagSettings
-      }
-    }, async () => {
-      try {
-        await syncItemsWithLocalStorage(userInfo)
-        await getSubscribedPodcasts()
-        await askToSyncWithNowPlayingItem()
-        await parseAllAddByRSSPodcasts()
-        await combineWithAddByRSSPodcasts()
-      } catch (error) {
-        console.log('loginUser setGlobal callback error:', error)
+    setGlobal(
+      {
+        session: {
+          userInfo: serverUserInfo,
+          isLoggedIn: true,
+          valueTagSettings
+        }
+      },
+      () => {
+        try {
+          const callback = async (newUserInfo: UserInfo) => {
+            await syncItemsWithLocalStorage(newUserInfo)
+            await getSubscribedPodcasts()
+            await askToSyncWithNowPlayingItem()
+            await parseAllAddByRSSPodcasts()
+            await combineWithAddByRSSPodcasts()
+          }
+          askToSyncLocalPodcastsWithServer(localUserInfo, serverUserInfo, callback)
+        } catch (error) {
+          console.log('loginUser setGlobal callback error:', error)
+        }
       }
-    })
-
-    return userInfo
+    )
   } catch (error) {
     throw error
   }
 }
 
+const askToSyncLocalPodcastsWithServer = (
+  localUserInfo: UserInfo, serverUserInfo: UserInfo, callback: any) => {
+  const localSubscribedPodcastIds = localUserInfo?.subscribedPodcastIds || []
+  const localAddByRSSPodcastFeedUrls = localUserInfo?.addByRSSPodcastFeedUrls || []
+  const serverSubscribedPodcastIds = serverUserInfo?.subscribedPodcastIds || []
+  const serverAddByRSSPodcastFeedUrls = serverUserInfo?.addByRSSPodcastFeedUrls || []
+
+  const unsavedSubscribedPodcastIds = localSubscribedPodcastIds?.filter(
+    (localId: string) =>
+      !serverSubscribedPodcastIds?.some((serverId: string) => serverId === localId)
+  )
+  const unsavedAddByRSSPodcastFeedUrls = localAddByRSSPodcastFeedUrls?.filter(
+    (localFeedUrl: string) =>
+      !serverAddByRSSPodcastFeedUrls?.some((serverFeedUrl: string) => serverFeedUrl === localFeedUrl)
+  )
+  
+  const handleSync = async () => {
```

Comment:
Since there is a ton of async stuff here that happens, will this block the UI if users hit yes and has a ton of local podcasts?

Reply:
I don't think so? The awaits are isolated to the handleSync method called by pressing "Yes, save them" in the alert. The user could still navigate around the app while the syncing happens...which might be non-ideal, but I don't think it should cause a serious problem.